app/queues/dns/dns-record-worker.server.ts

```diff
@@ -0,0 +1,211 @@
+
+export const dbRecordUpdateQueueName = 'db-record-update';
+export const dnsRecordUpdateQueueName = 'dns-record-update';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export enum WorkType {
+  create = 'create',
+  update = 'update',
+  delete = 'delete',
+}
+export interface JobRecord {
+  workType?: WorkType;
+  id?: Record['id'];
+  username: Record['username'];
+  type: Record['type'];
+  name: Record['name'];
+  value: Record['value'];
+}
+
+export type addDbRecordJobRecordResult = Record['id'];
+export type dnsRecordUpdateJobRecordResult = string;
+
+export const dbRecordWorker = new Worker<JobRecord>(
+  dbRecordUpdateQueueName,
+  async (job) => {
+    logger.debug('DB record update in DB');
+    const { workType, id, username, type, name, value } = job.data;
+
+    try {
+      switch (workType) {
+        case 'create':
+          const existingRecords = await prisma.record.count({
+            where: {
+              username,
+              type,
+              name,
+              value,
+            },
+          });
+
+          if (existingRecords) {
+            throw new Error('Record already exists');
+          }
+
+          const result = await prisma.record.create({
+            data: {
+              username,
+              type,
+              name,
+              value,
+              expiresAt: setMonthsFromNow(6),
+              status: RecordStatus.pending,
+            },
+          });
+          return result.id as addDbRecordJobRecordResult;
+        case 'update':
+          await prisma.record.update({
+            where: {
+              id,
+            },
+            data: {
+              type,
+              name,
+              value,
+              status: RecordStatus.pending,
+            },
+          });
+          break;
+        case 'delete':
+          await prisma.record.delete({
+            where: {
+              id,
+            },
+          });
+          break;
+        default:
+          throw new UnrecoverableError(`Invalid work type: ${workType}`);
+      }
+    } catch (error) {
+      logger.warn('Could not update record in DB', error);
+      throw error;
+    }
+  },
+  { connection: redis }
+);
+
+export const dnsRecordWorker = new Worker<JobRecord, dnsRecordUpdateJobRecordResult>(
+  dnsRecordUpdateQueueName,
+  async (job) => {
+    logger.debug('DNS record update in Route 53');
+    const { workType, username, type, name, value } = job.data;
+
+    try {
+      switch (workType) {
+        case 'create':
+          return createRecord(username, type, name.toLowerCase(), value);
+
+        case 'update':
+          return upsertRecord(username, type, name.toLowerCase(), value);
+
+        case 'delete':
+          return deleteRecord(username, type, name.toLowerCase(), value);
+
+        default:
+          throw new UnrecoverableError('Invalid work type');
```

Comment:
Same comment here: include the `workType` in the error message so it's easier to debug.